### ISSUE TITLE
module 'openai' has no attribute 'error'

### ISSUE THREAD

**Issue by ryanpeach** [langchain-ai/langchain#13368]
### System Info

openai==1.2.4
langchain==0.0.325
llama_index==0.8.69

### Who can help?

@hwchase17 @agola11 

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [X] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```
/usr/local/lib/python3.10/site-packages/llama_index/indices/base.py:102: in from_documents
    return cls(
/usr/local/lib/python3.10/site-packages/llama_index/indices/vector_store/base.py:49: in __init__
    super().__init__(
/usr/local/lib/python3.10/site-packages/llama_index/indices/base.py:71: in __init__
    index_struct = self.build_index_from_nodes(nodes)
/usr/local/lib/python3.10/site-packages/llama_index/indices/vector_store/base.py:254: in build_index_from_nodes
    return self._build_index_from_nodes(nodes, **insert_kwargs)
/usr/local/lib/python3.10/site-packages/llama_index/indices/vector_store/base.py:235: in _build_index_from_nodes
    self._add_nodes_to_index(
/usr/local/lib/python3.10/site-packages/llama_index/indices/vector_store/base.py:188: in _add_nodes_to_index
    nodes = self._get_node_with_embedding(nodes, show_progress)
/usr/local/lib/python3.10/site-packages/llama_index/indices/vector_store/base.py:100: in _get_node_with_embedding
    id_to_embed_map = embed_nodes(
/usr/local/lib/python3.10/site-packages/llama_index/indices/utils.py:137: in embed_nodes
    new_embeddings = embed_model.get_text_embedding_batch(
/usr/local/lib/python3.10/site-packages/llama_index/embeddings/base.py:250: in get_text_embedding_batch
    embeddings = self._get_text_embeddings(cur_batch)
/usr/local/lib/python3.10/site-packages/llama_index/embeddings/langchain.py:82: in _get_text_embeddings
    return self._langchain_embedding.embed_documents(texts)
/usr/local/lib/python3.10/site-packages/langchain/embeddings/openai.py:490: in embed_documents
    return self._get_len_safe_embeddings(texts, engine=self.deployment)
/usr/local/lib/python3.10/site-packages/langchain/embeddings/openai.py:374: in _get_len_safe_embeddings
    response = embed_with_retry(
/usr/local/lib/python3.10/site-packages/langchain/embeddings/openai.py:100: in embed_with_retry
    retry_decorator = _create_retry_decorator(embeddings)
/usr/local/lib/python3.10/site-packages/langchain/embeddings/openai.py:47: in _create_retry_decorator
    retry_if_exception_type(openai.error.Timeout)
E   AttributeError: module 'openai' has no attribute 'error'
```

### Expected behavior

I suppose it should run, I'll provide some reproducible code here in a minute.

**Comment by ryanpeach** [comment]
Very good summary @dosubot! I suspect this got missed by the rest of the team though. Its in the embeddings api which is less commonly used.

**Comment by ryanpeach** [comment]
Been particularly hard to track down a minimal reproducible example in my codebase. Let this thread know tomorrow if I discover one.

**Comment by EmiRoberti77** [comment]
Hello All, i had some problem and was reluctant to downgrade , so i tried first latest version available in case issue was fixed and problem seems to have been fixed on
openai==1.3.0
langchain==0.0.335

**Comment by dpromisel** [comment]
Here's a minimal example using:
```
pip install openai==0.28.1
pip install langchain==0.0.330
```

```
from openai import OpenAI
from langchain.document_loaders import PyPDFLoader
from langchain.vectorstores import FAISS
from langchain.embeddings.openai import OpenAIEmbeddings

loader = PyPDFLoader("./pdf.pdf")
pages = loader.load_and_split()

for page in pages:
    print(page.metadata) # metadata loads

faiss_index = FAISS.from_documents(pages, OpenAIEmbeddings())
docs = faiss_index.similarity_search("What is the primary purpose of this document?", k=5)
```

**Comment by DannyGsGit** [comment]
This does not appear to be fixed as of:
openai==1.3.6
langchain==0.0.343

The issue arises in Langchain's `embeddings.openai` in the [retry chunk](https://github.com/langchain-ai/langchain/blob/a14cf87576ef269346874cb9365fe5c2dac05066/libs/langchain/langchain/embeddings/openai.py#L53C1-L57C76).

`           retry_if_exception_type(openai.error.Timeout)
            | retry_if_exception_type(openai.error.APIError)
            | retry_if_exception_type(openai.error.APIConnectionError)
            | retry_if_exception_type(openai.error.RateLimitError)
            | retry_if_exception_type(openai.error.ServiceUnavailableError)
`
As of version 1.0.0, `openai` package now follows the convention `openai.Timeout`, `openai.RateLimitError`, etc... with no `.error`.

**Comment by zhifei92** [comment]
#### Same trouble
langchain              0.0.352
openai                 1.6.1
```
  File "/home/zhangzhifei/.local/lib/python3.10/site-packages/langchain_community/embeddings/localai.py", line 45, in _create_retry_decorator
    retry_if_exception_type(openai.error.Timeout)
AttributeError: module 'openai' has no attribute 'error'
```
"Is this related to the Python version, and are there any other approaches to consider?"

**Comment by PuTaTus** [comment]
I am facing the same issue.

AttributeError                            Traceback (most recent call last)

[<ipython-input-21-557f40a0267f>](https://localhost:8080/#) in <cell line: 72>()
     70 ]
     71 
---> 72 answer = chain.run({"input" : query, "chat_history": history})

14 frames

[/usr/local/lib/python3.10/dist-packages/langchain/chat_models/openai.py](https://localhost:8080/#) in _create_retry_decorator(self)
    300             wait=wait_exponential(multiplier=1, min=min_seconds, max=max_seconds),
    301             retry=(
--> 302                 retry_if_exception_type(openai.error.Timeout)
    303                 | retry_if_exception_type(openai.error.APIError)
    304                 | retry_if_exception_type(openai.error.APIConnectionError)

AttributeError: module 'openai' has no attribute 'error'

This is my dependencies:
Flask==2.3.2
Flask-Bcrypt==1.0.1
Flask-Cors==4.0.0
flask-redis==0.4.0
Flask-Session==0.5.0
langchain==0.0.335
langchainplus-sdk==0.0.20
numpy==1.25.1
openai==1.3.0
pymongo==4.5.0
python-dateutil==2.8.2
python-dotenv==1.0.0
requests==2.31.0

**Comment by s0l4r** [comment]
Also in: 
```langchain/chat_models/litellm.py", line 68, in _create_retry_decorator
    openai.error.Timeout,
    ^^^^^^^^^^^^```

AttributeError: module 'openai' has no attribute 'error'

openai = "1.6.1"
langchain = "0.0.325"

**Comment by fgabriel1891** [comment]
Hi. Same here. Run into the following: 

```
AttributeError                            Traceback (most recent call last)
[<ipython-input-16-ae0cc414eeda>](https://localhost:8080/#) in <cell line: 5>()
      3 chat = ChatOpenAI(openai_api_key= OPENAI_API_KEY, temperature = 0, cache=None, verbose=False )
      4 
----> 5 print(chat(msgs_suggest_questions))
      6 # Pass your message to GPT.
      7 

6 frames
[/usr/local/lib/python3.10/dist-packages/langchain/chat_models/openai.py](https://localhost:8080/#) in _create_retry_decorator(llm, run_manager)
     75 
     76     errors = [
---> 77         openai.error.Timeout,
     78         openai.error.APIError,
     79         openai.error.APIConnectionError,

AttributeError: module 'openai' has no attribute 'error'
```

**Comment by deniercounter** [comment]
@fgabriel1891 Usually this comes if you dont import openai.

Correct way:

```zsh
from openai import ChatOpenAI
import openai
```

**Comment by Bk073** [comment]
Did anyone solve it?

**Comment by munDane117** [comment]
thanks, it worked

> @fgabriel1891 Usually this comes if you dont import openai.
> 
> Correct way:
> 
> ```shell
> from langchain.chat_models import ChatOpenAI
> import openai
> ```

**Comment by Masoud-Karami** [comment]
[Here the issue is running](https://community.openai.com/t/getting-attributeerror-module-openai-has-no-attribute-error/578418/2)
You can find some alternatives based on your project. In my case, [this worked](https://community.openai.com/t/getting-attributeerror-module-openai-has-no-attribute-error/578418/11?u=massoud.karami).

**Comment by UsamaAmjad03** [comment]
> Hello All, i had some problem and was reluctant to downgrade , so i tried first latest version available in case issue was fixed and problem seems to have been fixed on openai==1.3.0 langchain==0.0.335

jeo shehzaday

**Comment by DJ-Manjaray** [comment]
Change your imports and decorator to this:

```
import backoff
from openai import APIError, RateLimitError
from openai import OpenAI

```

Then try:

```
@backoff.on_exception(backoff.expo, (RateLimitError, APIError), max_tries=5)
def safe_chat_completion(client: OpenAI, model: str, messages: list):
    return client.chat.completions.create(
        model=model,
        messages=messages,
        response_format={"type": "json_object"}
    )

```
